nextjsUIapp/src/app/knowledge-base/page.tsx

```diff
@@ -0,0 +1,184 @@
+        const query_to_execute = page ? page.sparql_query : "";
+        console.log(query_to_execute);
+        const page_title = page ? page.page : "";
+        const page_sub_title = page ? page.description : "";
+        setPageTitle(page_title);
+        setSubPageTitle(page_sub_title);
+
+        const queryParameter = {sparql_query: query_to_execute};
+
+        const baseurl = process.env.NEXT_PUBLIC_API_ADMIN_HOST || "http://3.134.90.242:8010";
+        const endpoint = process.env.NEXT_PUBLIC_API_QUERY_ENDPOINT || "query/sparql"; //default is "query/sparql"
+
+        console.log('Fetching data with parameters:', queryParameter, endpoint, baseurl);
+
+        try {
+            const response = await getData(queryParameter, endpoint, baseurl);
+            console.log('Raw response:', response);
+
+            if (response.status === 'success' && response.message?.results?.bindings) {
+                const bindings = response.message.results.bindings;
+                const vars = response.message.head.vars;
+                console.log('Bindings:', bindings);
+                console.log('Vars:', vars);
+                setHeaders(vars);
+                setData(bindings);
+            } else {
+                console.error('Unexpected response format:', response);
+                setError("Invalid data format");
+            }
+        } catch (e) {
+            const error = e as Error;
+            console.log('Error fetching data:', error.message);
+            setError(error.message);
+        } finally {
+            setLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        fetchData();
+    }, []);
+
+    const renderTable = () => {
+        if (!data || !Array.isArray(data) || data.length === 0) return null;
+
+        const startIndex = (currentPage - 1) * ITEMS_PER_PAGE;
+        const endIndex = startIndex + ITEMS_PER_PAGE;
+        const currentPageData = data.slice(startIndex, endIndex);
+
+        return (
+            <table className="w-full text-sm text-left rtl:text-right text-gray-500 dark:text-gray-400">
+                <thead
+                    className="sticky top-0 text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400">
+                <tr>
+                    {headers.map((header, index) => (
+                        <th key={index} className="border border-gray-400 px-4 py-2">{header}</th>
+                    ))}
+                </tr>
+                </thead>
+                <tbody>
+                {currentPageData.map((item, index) => (
+                    <tr key={index}
+                        className="bg-white border-b dark:bg-gray-800 dark:border-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600">
+                        {headers.map((header, headerIndex) => (
+                            <td key={headerIndex}
+                                className="px-6 py-4">{item[header]?.value.substring(item[header]?.value.lastIndexOf('/') + 1)}</td>
```

Comment:
Do you want a fallback value if item[header]? does not exist?

Reply:
> Do you want a fallback value if item[header]? does not exist?

No, at least not for the moment.